packages/ibm-products/src/components/Datagrid/Datagrid/addons/InlineEdit/InlineEditCell/InlineEditCell.js

```diff
@@ -401,9 +410,16 @@ export const InlineEditCell = ({
       <DatePicker
         {...datePickerPreparedProps}
         appendTo={outerButtonElement?.current?.parentElement}
-        ref={datePickerRef}
-        style={{
-          width: cell.column.totalWidth,
+        ref={(el) => {
+          datePickerRef.current = el;
+          if (datePickerRef.current && datePickerRef.current.style) {
+            datePickerRef.current.style.width = `${cell.column.totalWidth}px`;
+            const elementId = `${blockClass}__inline-edit--date-picker--${cell.row.index}`;
+            const element = document.getElementById(elementId);
```

Comment:
Is it possible to query this `element` from `el`, instead of finding from `document`?

Reply:
And if there are multiple `Datagrid` components on the same page, you're going to need to scope it to the correct datagrid instance.